File: PerfTester-Xamarin/PerfTester-Xamarin/Screens/TestList/TestListController.cs
using System;
using Foundation;
using PerfTesterXamarin.Screens.TestDetails;
using PerfTesterXamarin.Screens.TestList;
using PerfTesterXamarin.Screens.TestList.Cells;
using UIKit;

namespace PerfTesterXamarin
{
    public partial class TestListController : UIViewController
    {
        TestListViewModel ViewModel;
        public TestListController() : base("TestListController", null)
        {
            ViewModel = new TestListViewModel();
            SetupNavigationBar();
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();
            SetupTableView();
        }

		void SetupNavigationBar()
        {
            Title = "Tests";
        }

        void SetupTableView()
        {
            TableView.Delegate = new TestListDelegate(this, ViewModel);
            TableView.DataSource = new TestListDataSource(this, ViewModel);
            TableView.RowHeight = UITableView.AutomaticDimension;
            TableView.EstimatedRowHeight = 50;
            TableView.RegisterNibForCellReuse(TestListCell.Nib, "TestListCell");
        }
    }

    class TestListDataSource : UITableViewDataSource
    {
        WeakReference ListController;
        TestListViewModel ViewModel;
        public TestListDataSource(TestListController listController, TestListViewModel viewModel)
        {
            ListController = new WeakReference(listController);
            ViewModel = viewModel;
        }

        public override UITableViewCell GetCell(UITableView tableView, NSIndexPath indexPath)
        {
            var test = ViewModel.AllTests[indexPath.Row];
            TestListCell cell = (TestListCell)tableView.DequeueReusableCell("TestListCell");
            cell.SetupCellWithTest(test);
            return cell;
        }

        public override nint RowsInSection(UITableView tableView, nint section)
        {
            return ViewModel.AllTests.Length;
        }

    }

    class TestListDelegate : UITableViewDelegate
    {
        WeakReference ListController;
        TestListViewModel ViewModel;
        public TestListDelegate(TestListController listController, TestListViewModel viewModel)
        {
            ListController = new WeakReference(listController);
            ViewModel = viewModel;
        }

        public override void RowSelected(UITableView tableView, NSIndexPath indexPath)
        {
            if (!ListController.IsAlive) 
            {
                return;
            }

            var listCtr = ListController.Target as TestListController;

            var test = ViewModel.AllTests[indexPath.Row];
            var detailsCtr = new TestDetailsController(test);
            listCtr.NavigationController.PushViewController(detailsCtr, true);
        }

    }
}

File: PerfTester-Xcode/PerfTester-Xcode/Screens/TestList/TestListController.swift
//
//  TestListController.swift
//  PerfTester-Xcode
//
//  Created by Patryk Średziński on 08.06.2018.
//  Copyright © 2018 private.perftesterxcode. All rights reserved.
//

import UIKit

class TestListController: UIViewController {

    @IBOutlet fileprivate var tableView: UITableView!
    fileprivate let viewModel: TestListViewModel
    
    init() {
        viewModel = TestListViewModel();
        super.init(nibName: nil, bundle: nil)
        setupNavBar()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setupTableView()
    }

    private func setupNavBar() {
        self.title = "Tests"
    }
    
    func setupTableView() {
        tableView.delegate = self
        tableView.dataSource = self
        tableView.rowHeight = UITableViewAutomaticDimension
        tableView.estimatedRowHeight = 50
        tableView.register(UINib(nibName: "TestListCell", bundle: nil), forCellReuseIdentifier: "TestListCell")
    }
}

extension TestListController: UITableViewDataSource {
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.viewModel.allTests.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let test = self.viewModel.allTests[indexPath.row];
        if let cell = tableView.dequeueReusableCell(withIdentifier: "TestListCell") as? TestListCell {
            cell.setupCellWithTest(test);
            return cell;
        }
        return UITableViewCell()
    }
}

extension TestListController: UITableViewDelegate {
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let test = viewModel.allTests[indexPath.row]
        let detailsCtrl = TestDetailsController(test: test)
        navigationController?.pushViewController(detailsCtrl, animated: true)
    }
}
